Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -145,6 +148,73 @@ def render_PUT(self, request):
         return json.dumps({"added": channel_id})
 
 
+class MyChannelCreateTorrentEndpoint(MyChannelBaseEndpoint):
+    """
+    Create a torrent file from local files.
+    See: http://www.bittorrent.org/beps/bep_0012.html
+
+    Example request:
+    {
+        "files": ['path/to/file.txt', 'path/to/another/file.mp4'],
+        "description (optional)": "Video of my cat"
+        "trackers (optional)": [tracker1, backup1, backup2],
+    }
+    """
+
+    def render_GET(self, request):
+        """
+        This method creates a torrent from local files and returns it in base64 encoding.
+        """
+        parameters = http.parse_qs(request.content.read(), 1)
+        params = {}
+
+        if 'files' in parameters and len(parameters['files']) > 0:
+            file_path_list = ast.literal_eval(parameters['files'][0])
+        else:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.dumps({"error": "files parameter missing"})
+
+        if 'description' in parameters and len(parameters['description']) > 0:
+            params['comment'] = parameters['description'][0].encode('utf-8')
+
+        if 'trackers' in parameters and len(parameters['trackers']) > 0:
+            tracker_url_list = ast.literal_eval(parameters['trackers'][0])
+            params['announce'] = tracker_url_list[0]
+            params['announce-list'] = tracker_url_list
+
+        from Tribler.Core.version import version_id
+        params['created by'] = '%s version: %s' % ('Tribler', version_id)
+
+        params['nodes'] = False
+        params['httpseeds'] = False
+        params['encoding'] = False
+        params['piece length'] = 0  # auto
+
+        def _on_torrent_created(result):
+            """
+            Success callback
+            :param result: from create_torrent_file
+            """
+            with open(result['torrent_file_path'], 'rb') as f:
+                torrent_64 = base64.b64encode(f.read())
```

Comment:
Right now we have consistency between add-torrent-to-channel and create-torrent, they both use base64 encoding.
If you want to change that, change them both at the same time in a separate PR because add-torrent-to-channel is already merged.
Also, if you change it the pure JSON rest api is no longer JSON only.